Bulldozer.CSV/Maps/Attendance.cs

```diff
@@ -30,207 +33,328 @@ namespace Bulldozer.CSV
+                if ( occurrence != null )
+                {
+                    var newAttendance = new Attendance
+                    {
+                        PersonAliasId = attendanceImport.PersonAliasId,
+                        OccurrenceId = occurrence.Id,
+                        Note = attendanceImport.Note,
+                        DidAttend = true,
+                        ForeignId = attendanceImport.AttendanceForeignId,
+                        ForeignKey = attendanceImport.AttendanceForeignKey,
+                        CampusId = attendanceImport.CampusId,
+                        StartDateTime = attendanceImport.StartDateTime,
+                        EndDateTime = attendanceImport.EndDateTime,
+                        CreatedDateTime = importDateTime,
+                        ModifiedDateTime = importDateTime
+                    };
+                    attendancesToInsert.Add( newAttendance );
+                }
+            }
+            rockContext.BulkInsert( attendancesToInsert );
+            return attendancesToInsert.Count;
         }
 
         /// <summary>
-        /// Saves all group changes.
+        /// Create AttendanceOccurrences for AttendanceImports.
         /// </summary>
-        /// <param name="attendanceList">The attendance list.</param>
-        private static void SaveAttendance( List<Attendance> attendanceList )
+        /// <param name="rockContext">The Rock context.</param>
+        /// <param name="attendanceImports">The attendance imports.</param>
+        /// <param name="archivedAttendanceScheduleId">The ScheduleId for the default archived attendance schedule.</param>
+        /// <returns></returns>
+        public int CreateAttendanceOccurrences( RockContext rockContext, List<AttendanceImport> attendanceImports, Dictionary<string, Guid> existingOccurrenceGuidLookup, int archivedAttendanceScheduleId )
         {
-            //
-            // Save any attendance records
-            //
-            if ( attendanceList.Count > 0 )
+            int groupTypeIdFamily = GroupTypeCache.GetFamilyGroupType().Id;
+            var importDateTime = RockDateTime.Now;
+
+            // Create list of occurrences to be bulk inserted
+            var newOccurrences = new List<ImportOccurrence>();
+
+            var newAttendanceOccurrenceImports = attendanceImports
+                .Where( a => !existingOccurrenceGuidLookup.ContainsKey( a.OccurrenceForeignKey ) )
+                .GroupBy( a => a.OccurrenceForeignKey )
+                .Select( a => new
+                {
+                    OccurrenceForeignKey = a.Key,
+                    AttendanceImport = a.FirstOrDefault()
+                } )
+                .ToList();
+
+            foreach ( var attImport in newAttendanceOccurrenceImports )
             {
-                using ( var rockContext = new RockContext() )
+                var occurrence = new ImportOccurrence
```

Comment:
Is this interim object needed? Can we just directly create a list of AttendanceOccurrence objects here instead of below?